internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1634,6 +1645,7 @@ func resourceKubernetesClusterRead(d *pluginsdk.ResourceData, meta interface{})
 		d.Set("node_resource_group", props.NodeResourceGroup)
 		d.Set("enable_pod_security_policy", props.EnablePodSecurityPolicy)
 		d.Set("local_account_disabled", props.DisableLocalAccounts)
+		d.Set("public_network_access_enabled", props.PublicNetworkAccess == containerservice.PublicNetworkAccessEnabled)
```

Comment:
The API will return an empty string here for older clusters so this condition won't always return reflect the true state 
```suggestion
		d.Set("public_network_access_enabled", props.PublicNetworkAccess != containerservice.PublicNetworkAccessDisabled)
```